tests/Unit/Evolution/Ringdown/Test_StrahlkorperCoefsInRingdownDistortedFrame.cpp

```diff
@@ -0,0 +1,182 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Framework/TestingFramework.hpp"
+
+#include <array>
+#include <cmath>
+#include <cstddef>
+#include <limits>
+#include <random>
+#include <string>
+#include <vector>
+
+#include "DataStructures/DataVector.hpp"
+#include "Domain/CoordinateMaps/Distribution.hpp"
+#include "Domain/Creators/Sphere.hpp"
+#include "Domain/Creators/SphereTimeDependentMaps.hpp"
+#include "Domain/StrahlkorperTransformations.hpp"
+#include "Evolution/Ringdown/StrahlkorperCoefsInRingdownDistortedFrame.hpp"
+#include "Framework/TestHelpers.hpp"
+#include "Helpers/DataStructures/MakeWithRandomValues.hpp"
+#include "IO/H5/Dat.hpp"
+#include "IO/H5/File.hpp"
+#include "NumericalAlgorithms/SphericalHarmonics/IO/FillYlmLegendAndData.hpp"
+#include "NumericalAlgorithms/SphericalHarmonics/Spherepack.hpp"
+#include "NumericalAlgorithms/SphericalHarmonics/Strahlkorper.hpp"
+#include "PointwiseFunctions/AnalyticSolutions/GeneralRelativity/KerrHorizon.hpp"
+#include "Utilities/FileSystem.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/StdHelpers.hpp"
+
+// [[TimeOut, 10]]
+SPECTRE_TEST_CASE("Unit.Evolution.Ringdown.SCoefsRDis", "[Unit][Evolution]") {
+  // Write a temporary H5 file with Strahlkorpers at different times, then
+  // pass that file's path to strahlkorper_coefs_in_ringdown_distorted_frame().
+  // First, if the temporary file exists, remove it
+  const std::string horizons_file_name{"Unit.Evolution.Ringdown.SCoefsRDis.h5"};
+  const std::string horizons_subfile_name{"/ObservationAhC__Ylm.dat"};
+  if (file_system::check_if_file_exists(horizons_file_name)) {
+    file_system::rm(horizons_file_name, true);
+  }
+
+  MAKE_GENERATOR(generator);
+
+  // Start out with a Strahlkorper at rest in a grid frame, then map it
+  // to the inertial frame. The shape map is an identity map (it's initialized
+  // with Schwarzschild coefficients), so the grid->distorted map
+  // here is the identity. But start in a grid frame to use the available
+  // grid->distorted instantiation of strahlkorper_in_different_frame.
+  constexpr const std::array<double, 3> expected_center{4.4, 5.5, 6.6};
+  constexpr const size_t l_max = 12;
+  constexpr const size_t m_max = 12;
+  const auto kerr_horizon_radius = get(gr::Solutions::kerr_horizon_radius(
+      ::ylm::Spherepack(l_max, m_max).theta_phi_points(), 1.0,
+      {{0.0, 0.0, 0.8}}));
+  const auto expected_strahlkorper = ylm::Strahlkorper<Frame::Grid>(
+      l_max, m_max, kerr_horizon_radius, expected_center);
```

Comment:
If this is just supposed to be a Schwarzschild horizon of radius 1, you could do
```cpp
const ylm::Strahlkorper<Frame::Grid> expected_strahlkorper{
    l_max, 1.0, {0.0, 0.0, 0.0};
```

Reply:
I deliberately wanted it to be a Kerr horizon with spin 0.8, so some coefs are nontrivial

Reply:
Oh oops I misread this. The `8` blended in with all the `0`s